### DISCUSSION TITLE
Grace Period in subscription model

### DISCUSSION THREAD

**Discussion by cappy99** [Labs64/NetLicensing-Community#117] [Questions] [answered]
Where can I define how long the grace period is in the Subscription model? I am currently using community license.

**Comment by r-brown** [comment, marked as answer]
Hello @cappy99 you can enable the grace period for [Subscription](https://netlicensing.io/wiki/subscription) model on the module configuration page.
Simply enable "Grace Period" checkbox as follows:
<img width="1101" alt="Screenshot 2021-12-01 at 10 21 09" src="https://user-images.githubusercontent.com/1361258/144207893-908940f4-a21d-468d-9d49-1e06625da290.png">

**Comment by cappy99** [comment]
Thank you for your response. I understand how to turn on the grace period check box. 

However, this does not appear to change the "expires" property. How long is the grace period?  Can the grace time period be changed?  Do I need to upgrade my license to use this?

Thank you.

**Comment by r-brown** [comment]
There is no specific plan needed for grace period - this option is available for any time-volume licenses, such as Subscription.
Once you enabled a specific package as "grace", this will be added to the customer profile after main license expiration.
The length of the grace period is specified by the package marked as "Grace Period"

#### Product module configuration
<img width="1245" alt="grace-01" src="https://user-images.githubusercontent.com/1361258/144486922-4efb97a5-2911-4d5b-8883-72f7206e0966.png">

#### Grace Period license added
<img width="1088" alt="grace-02" src="https://user-images.githubusercontent.com/1361258/144486933-47888238-a131-44f9-a329-664c59a5343a.png">

---

... for instance

#### Validation response with the valid main license

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ns2:netlicensing xmlns="http://www.w3.org/2000/09/xmldsig#" xmlns:ns2="http://netlicensing.labs64.com/schema/context" ttl="2021-12-02T19:18:40.674Z">
  <ns2:infos/>
  <ns2:items>
    <ns2:item type="ProductModuleValidation">
      <ns2:property name="productModuleNumber">MC5KR0XW9-DEMO</ns2:property>
      <ns2:property name="valid">true</ns2:property>
      <ns2:property name="expires">2021-12-03T00:00:00Z</ns2:property>
      <ns2:property name="productModuleNumber">MC5KR0XW9-DEMO</ns2:property>
      <ns2:property name="productModuleName">Module using "Subscription" licensing model</ns2:property>
      <ns2:property name="warningLevel">YELLOW</ns2:property>
      <ns2:property name="licensingModel">Subscription</ns2:property>
    </ns2:item>
  </ns2:items>
</ns2:netlicensing>
```

#### Validation response with the activated grace license after main license expiration

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ns2:netlicensing xmlns="http://www.w3.org/2000/09/xmldsig#" xmlns:ns2="http://netlicensing.labs64.com/schema/context" ttl="2021-12-02T19:18:50.543Z">
  <ns2:infos>
    <ns2:info id="gracePeriodStarted" type="info">Subscription for product module 'MC5KR0XW9-DEMO' is extended with grace period.</ns2:info>
  </ns2:infos>
  <ns2:items>
    <ns2:item type="ProductModuleValidation">
      <ns2:property name="productModuleNumber">MC5KR0XW9-DEMO</ns2:property>
      <ns2:property name="valid">true</ns2:property>
      <ns2:property name="gracePeriod">true</ns2:property>
      <ns2:property name="expires">2021-12-03T18:18:50.543Z</ns2:property>
      <ns2:property name="productModuleName">Module using "Subscription" licensing model</ns2:property>
      <ns2:property name="warningLevel">YELLOW</ns2:property>
      <ns2:property name="licensingModel">Subscription</ns2:property>
    </ns2:item>
  </ns2:items>
</ns2:netlicensing>
```

**Comment by cappy99** [comment]
@r-brown thank you for the detailed response. This definitely answers my question. I was not including a separate grace package in my configuration.
